coral/src/app/features/topics/components/search/SearchTopics.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  Flexbox,
+  Icon,
+  InputBase,
+  PrimaryButton,
+} from "@aivenio/design-system";
+import { FormEvent, useState } from "react";
+import searchItem from "@aivenio/design-system/dist/module/icons/search";
+
+type SearchTopicsProps = {
+  searchTerm: string;
+  search: (searchTerm: string) => void;
+};
+
+function SearchTopics(props: SearchTopicsProps) {
+  const { search, searchTerm } = props;
+  const [currentSearchTerm, setCurrentSearchTerm] =
+    useState<string>(searchTerm);
+
+  function hasInput() {
```

Comment:
🙈  I avoid defining functions out of "read" order in React, so that I don't have a function name pop up without already having seen it. Also it messes the order up if there have to be some arrow functions, too, because then functions are located in two different places. 

I was used to and really liked the vue-specific: 

-> "markup"
-> props
-> state
-> life cycles
-> event handler functions
-> other functions

which would match your preference, right? But I don't think that's doable in react 🤔 

In React I usually worked in projects with the order:

-> props
-> state
-> life cycles
-> event handler functions
-> other functions
-> JSX

Which means the React FC only ends with a return statement, which matches what a "normal" function looks like.